test/unittests/libp2p/net.cpp

```diff
@@ -939,6 +940,85 @@ BOOST_AUTO_TEST_CASE(findNodeIsSentAfterPong)
     BOOST_CHECK_EQUAL(datagram3->typeName(), "FindNode");
 }
 
+BOOST_AUTO_TEST_CASE(pingNotSentAfterPongForKnownNode)
+{
+    // Validate 3 scenarios:
+    // * A ping is sent after a pong for a new node
+    // * No ping is sent after a pong for a known node with a valid endpoint proof
+    // * A ping is sent after a pong for a known node with an invalid endpoint proof
+
+    TestNodeTableHost nodeTableHost1(0);
+    nodeTableHost1.start();
+    auto& nodeTable1 = nodeTableHost1.nodeTable;
+
+    TestUDPSocketHost nodeSocketHost2{getRandomPortNumber()};
```

Comment:
small test simplification suggestion: I guess port parameter to `TestUDPSocketHost` is not needed anymore, as we always pass random port there (so we can just move `getRandomPortNumber()` inside constructor)

Reply:
I'll actually make the default value of the ctor's port parameter a call to getRandomPortNumber()...that way if we want to listen on a specific port for some reason we still can.